components/remerge/src/engine.rs

```diff
@@ -398,4 +398,50 @@ mod test {
         assert_eq!(um2["bar"], "test");
         um2.assert_tombstones(vec!["foo", "quux"]);
     }
+
+    #[test]
+    fn test_schema_cant_go_backwards() {
+        const FILENAME: &str = "test_schema_go_backwards.sqlite";
+        let _e: RemergeEngine = RemergeEngine::open(FILENAME, &*SCHEMA).unwrap();
+        let backwards_schema: String = json!({
+            "version": "0.1.0",
+            "name": "logins-example",
+            "fields": [],
+        })
+        .to_string();
+        let open_result = RemergeEngine::open(FILENAME, &*backwards_schema);
+        std::fs::remove_file(FILENAME).expect("uh oh, couldn't clean up test");
```

Comment:
I'm not sure of how best to clean up the SQLite databases when I'm done with them. This gets rid of the actual database but leaves `-shm` and `-wal` files around.

Reply:
RemergeEngine::open_in_memory is a thing, or creating it with the ":memory:" filename (https://www.sqlite.org/inmemorydb.html)

Reply:
How do I create an in-memory database that persists so I can access it again later?

Reply:
I think you can use a URI filename with a `mode=memory` param, like [here](https://www.sqlite.org/inmemorydb.html#sharedmemdb).

Reply:
Alternatively, if you can't make this work, you could put it inside the `target` directory, which will wipe it out with `cargo clean` and keep it from git status/etc

Be sure to delete it at the start of the test if you do this though.

Reply:
`mode=memory&cache=shared` worked like a charm, thanks @linacambridge!